src/tribol/mesh/CouplingScheme.cpp

```diff
@@ -1482,46 +1467,50 @@ void CouplingScheme::computeCommonPlaneTimeStep(real &dt)
+         // gap projected onto the outward unit face-normal to check against the
+         // max allowable gap as a factor of the thickness in the element normal
+         // direction 
          real delta1 = max_delta1 - gap_f1_n1; // >0 not exceeding max allowable
          real delta2 = max_delta2 + gap_f2_n2; // >0 not exceeding max allowable
 
-         if (delta1 < 0 || delta2 < 0)
-         {
-            max_gap_msg = true;
-         }
+         exceed_max_gap1 = (delta1 < 0.) ? true : false;
+         exceed_max_gap2 = (delta2 < 0.) ? true : false;
 
          // if velocity projection indicates further interpenetration, and the gaps
          // EXCEED max allowable, then compute time step estimates to reduce overlap
-         dt1_check1 = (dt1_vel_check) ? (delta1 < 0.) : false;
-         dt2_check1 = (dt2_vel_check) ? (delta2 < 0.) : false;
+         dt1_check1 = (dt1_vel_check) ? exceed_max_gap1 : false;
+         dt2_check1 = (dt2_vel_check) ? exceed_max_gap2 : false;
 
-         // compute dt for face 1 and 2 based on the velocity projection in the 
-         // direction of that face's outward unit normal
-         // Note, this calculation takes a fraction of the computed dt to reduce 
-         // the amount of face-displacement in a given cycle.
+         // compute dt for face 1 and 2 based on the velocity and gap projections onto 
+         // the face-normals for faces where currect gap exceeds max allowable gap.
+         //
+         // NOTE:
          //
-         // if dt[i]_check[i] is true, then delta[i] is < 0. per check above. Furthermore,
-         // if the velocity projection indicates further interpenetration, the velocity 
-         // projected onto that face's outward unit normal is always positive. Thus,
-         // dt[i] should never be negative unless the face-normal is flipped based on 
-         // vertex ordering.
-         dt1 = (dt1_check1) ? -alpha * delta1 / v1_dot_n1 : dt1;
-         dt2 = (dt2_check1) ? -alpha * delta2 / v2_dot_n2 : dt2;
-
-         //std::cout << "dt1_check1, delta1 and v1_dot_n1: " << dt1_check1 << ", " << delta1 << ", " << v1_dot_n1 << std::endl;
-         //std::cout << "dt2_check1, delta2 and v2_dot_n2: " << dt2_check1 << ", " << delta2 << ", " << v2_dot_n2 << std::endl;
+         // This calculation RESETS the current gap to be g = 0, and computes a timestep
+         // such that the velocity projection of the overlap-to-face projected overlap 
+         // centroid does not exceed the max allowable gap. 
+         //
+         // This avoid a timestep crash in the case that the current gap barely exceeds 
+         // the max allowable and also allows a soft contact response with interpen
+         // in excess of the max allowable gap without causing timestep crashes.
+         //
+         // v1_dot_n1 > 0 and v2_dot_n2 > 0 for further interpen
+         dt1 = (dt1_check1) ? alpha * max_delta1 / v1_dot_n1 : dt1;
```

Comment:
Can we expose alpha to the host code?  This will allow me to test the timestep vote in ALE3D in the way that we would actually use it.

Reply:
Added. Default is 1.0. Alpha defaults to 1.0 if user specifies alpha <= 0. Let's chat about how you would use this in your host code so I can apply this logic to mine.